lib/Twig/Extensions/Node/Trans.php

```diff
@@ -41,6 +41,10 @@ public function compile(Twig_Compiler $compiler)
 
         $function = null === $this->getNode('plural') ? 'gettext' : 'ngettext';
 
+        if (null !== $this->getNode('domain')) {
```

Comment:
So the problem is that the value of domain node is actually used. Will a simple if in the compiled code solve the problem?